### PR TITLE
Viresh_Assignment_29_Done_40Mins

### PR DESCRIPTION
**Pull request by VireshJJun20**
Please review and comment, thanks.

image

### CODE REVIEW

File: src/vireshJJun20/ConvertCase.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+public class ConvertCase {
+
+	static String tempString2 = "";
+
+	// Method accepts Array, converts each index to appropriate case and then saves
+	// in output array.
+	String[] StringConvertCase(String[] inputArray) {
+		String[] outputArray = inputArray;
+		for (int arrIndex = 0; arrIndex < inputArray.length; arrIndex++) {
+			String str = inputArray[arrIndex];
```
Comment (DeavinaSJun20) [inline src/vireshJJun20/ConvertCase.java]: we can remove the **str** variable here and just apply the **charAt** function on **inputArray[index].**

File: src/vireshJJun20/ConvertCase.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+			String tempString = "";
+			for (int strIndex = 0; strIndex < str.length(); strIndex++) {
+
+				if (str.charAt(strIndex) >= 65 && str.charAt(strIndex) <= 90) {
+					tempString += (char) ((int) str.charAt(strIndex) + 32);
+				} else if (str.charAt(strIndex) >= 'a' && str.charAt(strIndex) <= 'z') {
+					tempString += (char) ((int) str.charAt(strIndex) - 32);
+				} else {
+					tempString2 += str.charAt(strIndex) + " ";
```
Comment (DeavinaSJun20) [inline src/vireshJJun20/ConvertCase.java]: why are we using the second else. could you please explain?

File: src/vireshJJun20/ConvertCase.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+		String[] inputArr = { "AaVruti", "NiKhil", "AdiTi", "MAhesh", "ArchaNA", "jaGAdeesh" };
+		System.out.print("Input Array : (");
+		new ConvertCase().displayArray(inputArr);
+		String[] outputArray = new ConvertCase().StringConvertCase(inputArr);
+		System.out.println("");
+		System.out.print("Output Array: (");
+		new ConvertCase().displayArray(outputArray);
+		System.out.println("");
+		if (!(tempString2.isEmpty())) {
+			System.out.println("Array contains few Strings with special chars/integers which cant be converted to either upper case/lower case.");
```
Comment (DeavinaSJun20) [inline src/vireshJJun20/ConvertCase.java:63]: Also please explain the use for this loop

**Review by DeavinaSJun20 — CHANGES_REQUESTED** [review]
Please resolve the comments

File: src/vireshJJun20/ConvertCase.java (hunk 1 above)
Comment (VireshJJun20) [reply src/vireshJJun20/ConvertCase.java]: Yes we can, however I had used it for more readability.

File: src/vireshJJun20/ConvertCase.java (hunk 2 above)
Comment (VireshJJun20) [reply src/vireshJJun20/ConvertCase.java]: This is used to capture special character Or a digit in an array element.

File: src/vireshJJun20/ConvertCase.java (hunk 3 above)
Comment (VireshJJun20) [reply src/vireshJJun20/ConvertCase.java:63]: If you are asking about the if condition, its just a display message if the Array element contains any special character or any digit.

File: src/vireshJJun20/ConvertCase.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+package vireshJJun20;
+
+public class ConvertCase {
+
+	static String tempString2 = "";
+
+	// Method accepts Array, converts each index to appropriate case and then saves
+	// in output array.
+	String[] StringConvertCase(String[] inputArray) {
+		String[] outputArray = inputArray;
```
Comment (KrishnaKTechnocredits) [inline src/vireshJJun20/ConvertCase.java:22]: If you are creating a new array then you should create till it's length String[] outputArray = new String[inputArray.length], here you are assigning old array in new then what's the difference?

**Pull request merged**